### PR TITLE
docs: Improve GitHub Actions snippets

### PR DESCRIPTION
**Pull request by carloscuesta**
Hello! 👋🏼 

Yesterday I was migrating from Travis-CI to GitHub Actions and I found a couple of things that are not mentioned on the docs that could be helpful to include

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,10 +18,14 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```
Comment (carloscuesta) [inline README.md:22]: This is related with the following issue: https://github.com/GoogleChrome/lighthouse-ci/issues/172

File: README.md [hunk 2]
```diff
@@ -18,10 +18,14 @@ jobs:
...
       - uses: actions/setup-node@v1
       - run: npm install && npm install -g @lhci/cli@0.6.x
       - run: npm run build
       - run: lhci autorun
+        env:
+          LHCI_GITHUB_APP_TOKEN: ${{ secrets.LHCI_GITHUB_APP_TOKEN }}
```
Comment (carloscuesta) [inline README.md:28]: This is mentioned on the docs, that you should add the LHCI_GITHUB_APP_TOKEN as an environment variable on the repository but you should include as env inside of the action

**Review by patrickhulce — CHANGES_REQUESTED** [review]
Thanks for the feedback and improvements @carloscuesta! 🎉 

These two changes are both specific to GitHub status checks and we've tried quite intentionally to avoid including too many future steps in the early parts of the documentation (like the quick start and step 1 of getting started that are changed here.

What do you think of adding this instead under "GitHub Status Checks" section with a special note about ensuring git history is available in *all* CI providers and providing the example for GitHub actions like you've done?

**Comment by carloscuesta** [comment]
@patrickhulce I think it makes sense! I'm making the updates you suggested, thanks for taking the time to review the PR 😊

**Review by patrickhulce — APPROVED** [review]
thanks @carloscuesta!

**Comment by carloscuesta** [comment]
You're welcome! Happy to contribute 😊

**Pull request merged**